roles/feature-netapp-trident/tasks/post-deploy.yaml

```diff
@@ -0,0 +1,218 @@
+    state: present
+    definition:
+      apiVersion: snapshot.storage.k8s.io/v1
+      kind: VolumeSnapshotClass
+      metadata:
+        name: coe-netapp
+      driver: csi.trident.netapp.io
+      deletionPolicy: Delete
+
+- name: Create secret with NetApp credentials
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cluster_access_kubeconfig }}"
+    state: present
+    definition:
+      apiVersion: v1
+      kind: Secret
+      metadata:
+        name: coe-netapp-svm-trident
+        namespace: netapp-trident
+      type: Opaque    
+      data:
+        password: "{{ netapp_password | b64encode }}"
+        username: "{{ netapp_username | b64encode }}"
+
+- name: Create TridentBackendConfig/coe-netapp-san
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cluster_access_kubeconfig }}"
+    state: present
+    definition:
+      apiVersion: trident.netapp.io/v1
+      kind: TridentBackendConfig
+      metadata:
+        name: coe-netapp-san
+        namespace: netapp-trident
+      spec:
+        version: 1
+        storageDriverName: ontap-san-economy
+        managementLIF: "{{ netapp_managementLIF }}"
+        backendName: coe-netapp-san
+        storagePrefix: "stormshift_{{ inventory_hostname }}_"
+        credentials:
+          name: coe-netapp-svm-trident
+
+- name: Create TridentBackendConfig/coe-netapp-nas
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cluster_access_kubeconfig }}"
+    state: present
+    definition:
+      apiVersion: trident.netapp.io/v1
+      kind: TridentBackendConfig
+      metadata:
+        name: coe-netapp-nas
+        namespace: netapp-trident
+      spec:
+        version: 1
+        storageDriverName: ontap-nas
+        managementLIF: "{{ netapp_managementLIF }}"
+        backendName: coe-netapp-nas
+        storagePrefix: "stormshift_{{ inventory_hostname }}_"
+        credentials:
+          name: coe-netapp-svm-trident
+
+- name: Create StorageClass/coe-netapp-san
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cluster_access_kubeconfig }}"
+    state: present
+    definition:
+      apiVersion: storage.k8s.io/v1
+      kind: StorageClass
+      metadata:
+        annotations:
+          storageclass.kubevirt.io/is-default-virt-class: "true"
+        name: coe-netapp-san
+      provisioner: csi.trident.netapp.io
+      parameters:
+        backendType: "ontap-san-economy"
+        fsType: "ext4"
+      mountOptions:
+        - discard
+      allowVolumeExpansion: true
+
+- name: Create StorageClass/coe-netapp-nas
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cluster_access_kubeconfig }}"
+    state: present
+    definition:
+      apiVersion: storage.k8s.io/v1
+      kind: StorageClass
+      metadata:
+        annotations:
+          storageclass.kubernetes.io/is-default-class: "true"
+        name: coe-netapp-nas
+      provisioner: csi.trident.netapp.io
+      parameters:
+        backendType: "ontap-nas"
+        provisioningType: "thin"
+        snapshots: "true"
+
```

Comment:
Maybe add some wait task here for everything to stabilize and be ready? I could image that this feature is one of the first in the sequence, because subsequent feature like RHOAI need storage, and it would be nice if those then do not fail.  Eg. wait for the operator pod to be ready, or some of the Trident CRs maybe report a nice status we could wait for to be reached?